### PR TITLE
cli: config: leverage toml_edit::Value to serialize values

### PR DESCRIPTION
**Pull request by yuja**
#3374

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

**Review by bnjmnt4n — APPROVED** [review]

**Review by martinvonz — APPROVED** [review]
Thanks!

**Comment by ilyagr** [comment]
You could use the test from https://github.com/martinvonz/jj/pull/3731/files#diff-4256ae2574c92c8fe81a570c6f4df6b91d19cbe3465cb1b10560986407c8ca83.

File: cli/src/commands/config.rs [hunk 1]
```diff
@@ -174,6 +173,113 @@ fn toml_escape_key(key: String) -> String {
     toml_edit::Key::from(key).to_string()
 }
 
+fn to_toml_value(value: &config::Value) -> Result<toml_edit::Value, config::ConfigError> {
```
Comment (ilyagr) [inline cli/src/commands/config.rs:176]: (No action needed) I'm not sure why `config::Value` does not implement `Serialize`. I thought for a second of contributing it, but it seems the maintainer of `config` is already overwhelmed.

File: cli/tests/test_config_command.rs [hunk 2]
```diff
@@ -108,7 +108,7 @@ fn test_config_list_inline_table() {
     );
     let stdout = test_env.jj_cmd_success(test_env.env_root(), &["config", "list", "test-table"]);
     insta::assert_snapshot!(stdout, @r###"
-    test-table=[{x=1}, {y=["z"]}]
+    test-table=[{ x = 1 }, { y = ["z"] }]
```
Comment (ilyagr) [inline cli/tests/test_config_command.rs:111]: You could use the test from https://github.com/martinvonz/jj/pull/3731/files#diff-4256ae2574c92c8fe81a570c6f4df6b91d19cbe3465cb1b10560986407c8ca83.

**Comment by ilyagr** [comment]
In my version of this PR (I didn't notice yours :( ), I added a not to the CHANGELOG: https://github.com/martinvonz/jj/pull/3731/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed. Feel free to take it or leave it.

**Comment by ilyagr** [comment]
I also wonder whether we should care about letting people get some info from the output without a full TOML parser.

We could have an option --simple that prints each option in one line and maybe replaced any `=` signs in keys with `EQUALS`. Then you can easily get a representation of the key and value you could show to the user.

**Review by yuja — COMMENTED** [review]
> We could have an option `--simple` that prints each option in one line and maybe replaced any `=` signs in keys with `EQUALS`. Then you can easily get a representation of the key and value you could show to the user.

Perhaps, it's the job of templating, something like `value.to_string().replace(..)`.

**Comment by ilyagr** [comment]
> Perhaps, it's the job of templating, something like value.to_string().replace(..).

This won't work if the value is a multi-lined `'''` string ("Multi-line literal string" in https://toml.io/en/v1.0.0#string).

**Comment by yuja** [comment]
> > Perhaps, it's the job of templating, something like value.to_string().replace(..).
> 
> This won't work if the value is a multi-lined `'''` string ("Multi-line literal string" in https://toml.io/en/v1.0.0#string).

I mean `value` here is an enum of `bool|i64|string|..`, and `.to_string()` will extract it as string (not serialized blob.)

**Pull request merged**